tests/FeeSharingCollectorTest.js

```diff
@@ -2064,6 +2064,550 @@ contract("FeeSharingCollector:", (accounts) => {
+            /// @dev This test requires redeploying the protocol
+            await protocolDeploymentFixture();
+
+            const checkpoints = 10;
+            const maxCheckpoint = 3;
+
+            // stake - getPriorTotalVotingPower
+            await stake(900, root);
+            let userStake = 100;
+            if (MOCK_PRIOR_WEIGHTED_STAKE) {
+                await staking.MOCK_priorWeightedStake(userStake * 10);
+            }
+            await SOVToken.transfer(account1, userStake);
+            await stake(userStake, account1);
+
+            // mock data
+            await createCheckpoints(checkpoints);
+
+            await mine(2880 * 15, { interval: 30 }); // 86400 (1day) / 30 == 2800 * 15 (2 weeks + 1 day - for weighted stake to be updated in cache of FeeSharingCollector._getAccumulatedFees())
+
+            const totalCheckpoints = await feeSharingCollector.totalTokenCheckpoints(
+                RBTC_DUMMY_ADDRESS_FOR_CHECKPOINT
+            );
+            let iteration = 0;
+
+            for (let i = 0; i < totalCheckpoints; i += maxCheckpoint) {
+                iteration += 1;
+            }
+
+            let allUserFees = await feeSharingCollector.getAllUserFeesPerMaxCheckpoints(
+                account1,
+                RBTC_DUMMY_ADDRESS_FOR_CHECKPOINT,
+                maxCheckpoint
+            );
+
+            expect(allUserFees["nextCheckpointNum"]).to.equal(0);
+            expect(allUserFees["fees"].length).to.equal(iteration);
+
+            let feesIndex = 0;
+            for (let i = 0; i < totalCheckpoints; i += maxCheckpoint) {
+                const fees = await feeSharingCollector.getAccumulatedFeesForCheckpointsRange(
+                    account1,
+                    RBTC_DUMMY_ADDRESS_FOR_CHECKPOINT,
+                    i,
+                    maxCheckpoint
+                );
+                expect(allUserFees["fees"][feesIndex].toString()).to.equal(fees.toString());
+                feesIndex++;
+            }
+
+            let tx = await feeSharingCollector.withdrawRbtcTokens(
+                [RBTC_DUMMY_ADDRESS_FOR_CHECKPOINT],
+                2,
+                ZERO_ADDRESS,
+                {
+                    from: account1,
+                }
+            );
+
+            console.log("\nwithdraw(checkpoints = 10).gasUsed: " + tx.receipt.gasUsed);
+            // processedCheckpoints
+            let processedCheckpoints = await feeSharingCollector.processedCheckpoints.call(
+                account1,
+                RBTC_DUMMY_ADDRESS_FOR_CHECKPOINT
+            );
+            expect(processedCheckpoints.toNumber()).to.be.equal(2);
+
+            allUserFees = await feeSharingCollector.getAllUserFeesPerMaxCheckpoints(
+                account1,
+                RBTC_DUMMY_ADDRESS_FOR_CHECKPOINT,
+                100
+            );
+
+            expect(allUserFees["nextCheckpointNum"]).to.equal(2);
```

Comment:
why fees amounts are not checked after withdrawal here and in other tests?

Reply:
The fee is already checked here in the below
https://github.com/DistributedCollective/Sovryn-smart-contracts/blob/a256bcebe429bf392b7c3f68fcffc272525c2163/tests/FeeSharingCollectorTest.js#L2322C1-L2323C1